model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java

```diff
@@ -739,7 +740,17 @@ protected void removeUserSession(UserSessionEntity sessionEntity, boolean offlin
     UserSessionAdapter wrap(RealmModel realm, UserSessionEntity entity, boolean offline) {
         InfinispanChangelogBasedTransaction<String, UserSessionEntity> userSessionUpdateTx = getTransaction(offline);
         InfinispanChangelogBasedTransaction<UUID, AuthenticatedClientSessionEntity> clientSessionUpdateTx = getClientSessionTransaction(offline);
-        return entity != null ? new UserSessionAdapter(session, this, userSessionUpdateTx, clientSessionUpdateTx, realm, entity, offline) : null;
+
+        if (entity == null) {
+            return null;
+        }
+
+        UserModel user =  session.users().getUserById(realm, entity.getUser());
+        if (user == null) {
+            return null;
```

Comment:
Should we remove the user session before we return `null`?
```suggestion
            removeUserSession(entity, offline);
            return null;
```

Reply:
I think the session removal should not be done.
I have the following reasons:
- The session will time out sooner or later anyway.
- Especially with federated users, it could be a temporary issue. A later lookup could return a user. With offline sessions not being invalidated here lots of users would not need to login again but just wait until the temporary issue is solved.

Reply:
That sounds reasonable @mkrueger92, thanks!